bsip-0044.md

```diff
@@ -0,0 +1,173 @@
+
+### **Early Termination of an Escrow Contract**
+
+To protect the `recipient`, early termination of an HTLC is not allowed by any party. Placing a `timeout threshold` far into the future is valid. User protection from locking up funds for an extremely long period could be provided by the UI used to create the HTLC.
+
+### **Automatic Transfers Upon Expiry**
+
+Upon expiry of the `timeout threshold`, the `escrow assets` held within the HTLC will **not** automatically be returned to `depositor`. A separate transaction must be initiated by the `depositor` to request `escrow assets` be transferred to `depositor`. 
+
+Code _could_ be added to automate the return of funds. This could be part of block production or maintenance. However, this will cause extra load on those processes. Hence, it is felt that requiring action by the `depositor` is the best option.
+
+### **Fee**
+
+Creating and fulfillment are two operations that add data to the blockchain. The `fee` for each operation is based on the standards set for blocks, and is similar to costs of other items stored on-chain.
+
+## **Existing Escrow Proposals**
+
+This section describes various escrow concepts that have been proposed either for BitShares or for other blockchains or services in terms of the elements that have been defined above. This is intended to provide some background and comparison to the concepts that follow.
+
+## **Existing Graphene Features that are Similar to What is Needed for Blockchain Escrowing**
+
+### **BitShares Escrow**
+
+A separate BSIP [cite] is currently being discussed that provides a more traditional escrow service. This involves parties, agents, and a more complex evaluation. HTLC shares some similarities, and could be considered a thin subset of BitShares Escrow.
+
+The smaller, well-defined nature of HTLC provides a major advantage for applications that want to attempt tasks such as cross chain atomic swaps.
+
+### **BitShares Multi-Signature Account**
+
+One of the existing features of BitShares is the ability to have an account that requires multiples signatures by differently authorized parties [cite] and even hierarchical authorizations. Using this mechanism as a form of escrow is possible. But there are many limitations. More information on escrow and multi-signatures can be found in the BitShares Escrow BSIP.
+
+### **BitShares Proposals**
+
+One of the existing features of BitShares is the ability to have a proposal that is recorded on the blockchain and awaits the authorization of the requisite parties (e.g. M-of-N signatures). If the required authorizations are not given by the proposal expiry then no transfer will occur. This feature also contains many limitations when compared to HTLC.
+
+# **Specifications**
+
+## **Possible Concepts to Implement**
```

Comment:
These concepts are possible applications of the proposed functionality and belong into the "Motivation" section.

The "Specification"  section should contain a detailed description of the proposed protocol changes, i. e. operations to be added, evaluation logic to be added, other supporting changes (like adding new fee types and extending global parameters accordingly).

Reply:
Got it. Moving Specifications after these examples and will add details thereto.